include/wlr/types/wlr_keyboard.h

```diff
@@ -45,12 +52,7 @@ struct wlr_keyboard {
 
 	uint32_t keycodes[WLR_KEYBOARD_KEYS_CAP];
 	size_t num_keycodes;
-	struct {
-		xkb_mod_mask_t depressed;
-		xkb_mod_mask_t latched;
-		xkb_mod_mask_t locked;
-		xkb_mod_mask_t group;
-	} modifiers;
+	struct wlr_keyboard_modifiers *modifiers;
```

Comment:
No need to use a pointer here.

Reply:
Maybe you're right, but why not use a pointer?

Reply:
Because you have to allocate another data structure, check if it fails, free it. When accessing fields there's an extra pointer indirection (and maybe a cache miss). Here we don't really care about perf, but it's just more readable, easier and simpler.